tutorials/add_model.md

```diff
@@ -0,0 +1,140 @@
+# Contributing a model to Cornac
+
+This tutorial describes how to integrate a recommender model into Cornac. We assume that you have already forked the Cornac repository to your own account.
+
+## Directory & file structure
+
+For convenience assume that the model of interest is PMF. At this point it is useful to recall that Cornac follows the [PEP 8](https://www.python.org/dev/peps/pep-0008/) coding conventions. One good way to get started is to update the directory and file structure as follows.
+```
+cornac    
+│
+└───cornac
+│   │
+│   └───models
+│       │   __init__.py
+│       │   recommender.py
+│       │
+│       └───pmf
+│           │   __init__.py
+│           │   recom_pmf.py
+│           │   requirements.txt
+```
+Note that you only need to add the `pmf` branch as the rest of the structure is already in place.
+
+## Creating a model in 4 steps
+
+### 1. Extending the Recommender class
+
+The main file is `recom_pmf.py`, which will contain your recommender algorithm related codes. The name of this file should always start with the prefix `recom_`. Here is the minimal structure of such file:
+```python
+from ..recommender import Recommender
+
+class PMF(Recommender):
+    """Probabilistic Matrix Factorization.
+
+    Parameters
+    ----------
+
+    References
+    ----------
+    """
+
+    def __init__(self, name="PMF", trainable=True, verbose=False, ...):
+        Recommender.__init__(self, name=name, trainable=trainable, verbose=verbose)
+
+    def fit(self, train_set):
+        """Fit the model to observations.
+
+        Parameters
+        ----------
+        train_set: object of type TrainSet, required
+            An object contraining the user-item preference in csr scipy sparse format,\
+            as well as some useful attributes such as mappings to the original user/item ids.\
+            Please refer to the class TrainSet in the "data" module for details.
+        """
+
+    def score(self, user_id, item_id=None):
+        """Predict the scores/ratings of a user for an item.
+
+        Parameters
+        ----------
+        user_id: int, required
+            The index of the user for whom to perform score prediction.
+
+        item_id: int, optional, default: None
+            The index of the item for that to perform score prediction.
+            If None, scores for all known items will be returned.
+
+        Returns
+        -------
+        res : A scalar or a Numpy array
+            Relative scores that the user gives to the item or to all known items
+        """
+``` 
+Every model is an extension of the generic class `Recommender`. All you need to do is to redefine the functions listed in the above `recom_pmf.py` file.
```

Comment:
I think we should avoid the word `extension` here because we have Cython extensions. Furthermore, inheritance is not something bad to mention here.

Reply:
I can improve it, yet it is clear from the context we are referring to inheritance and there is no ambiguity.